app/src/common/main.c

```diff
@@ -31,38 +37,84 @@ static void app_exit(void) {
     END_TRY_L(exit);
 }
 
+// Helper to handle the different library commands
+static void library_main(libargs_t *args) {
+    BEGIN_TRY {
+        TRY {
+            switch (args->command) {
+                case SIGN_TRANSACTION: {
+                    // Backup up transaction parameters and wipe BSS to avoid collusion with
+                    // app-exchange BSS data.
+                    bool success = copy_transaction_parameters(args->create_transaction);
+                    if (success) {
+                        // BSS was wiped, we can now init these globals
+                        G_swap_state.called_from_swap = true;
+                        G_swap_state.should_exit = false;
+
+#ifdef HAVE_NBGL
+                        // On Stax, display a modal
+                        nbgl_useCaseSpinner("Signing");
```

Comment:
Internal note: we can add in zxlib a spinner/loading interface to avoid using UI methods directly